pkg/data/egress_lists/egress_lists.go

```diff
@@ -46,18 +87,14 @@ func GetLocalEgressList(platformType cloud.Platform) (string, error) {
 	case cloud.AWSHCPZeroEgress:
 		return templateAWSHCPZeroEgress, nil
 	default:
-		return "", fmt.Errorf("no egress list registered for platform '%s'", platformType)
+		return "", fmt.Errorf("no egress list registered for platform '%s'", g.PlatformType)
 	}
 }
 
-func GetGithubEgressList(platformType cloud.Platform) (*github.RepositoryContent, error) {
-	ghClient := github.NewClient(nil)
+func (g *Generator) GetGithubEgressList(ctx context.Context) (*github.RepositoryContent, error) {
 	path := "/pkg/data/egress_lists/"
-	if !platformType.IsValid() {
-		fmt.Printf("Platform type %s is invalid", platformType)
```

Comment:
As described in the PR body, I removed this check, which was only printing the invalid platform and then continuing execution.

Reply:
Do we want to change this to short circuit the execution at this point if the platform type is invalid?  Worth noting there may be earlier checks before it gets this far that would short circuit already

Reply:
We provide the valid platform types as global variables, and the method `ByName(name string) (string, error)`, so it feels unnecessary to also then later validate a `Platform` is valid since callers should be utilizing the variables or `ByName...` to construct the type. This is further evident by the `names` field of the struct being private so it wouldn't be possible to construct an invalid one anyways afaict.

Reply:
> it wouldn't be possible to construct an invalid one anyways afaict
There are several points in the codebase where we return an empty (and therefore invalid) `Platform{}`

Reply:
> > it wouldn't be possible to construct an invalid one anyways afaict
> 
> There are several points in the codebase where we return an empty (and therefore invalid) `Platform{}`

Based on my searching, this is only happening under error cases inside other functions and in tests. If an upstream caller is not handling that error, and then using the `Platform{}` then that is a bug that should be fixed. Was there another example you had of this (outside of tests and errors)?

Reply:
Correct, there are currently no other cases currently where `Platform{}` is used in network-verifier directly but it is possible to return an invalid type (with error) through `platform.ByName()` or `platform.IsValid()`.  As long as upstream callers handle that error I'm fine conceding the point and don't think the check is necessary here.  As far as I can tell the `ValidateEgress` func handles that beforehand.

Reply:
> Correct, there are currently no other cases currently where `Platform{}` is used in network-verifier directly but it is possible to return an invalid type (with error) through `platform.ByName()` or `platform.IsValid()`. As long as upstream callers handle that error I'm fine conceding the point and don't think the check is necessary here. As far as I can tell the `ValidateEgress` func handles that beforehand.

Totally, and yup we handle the error validation already [here](https://github.com/openshift/osd-network-verifier/blob/2b020def9f2d5c2f4068d5d2804c6496c76795ce/pkg/verifier/aws/entry_point.go#L37-L39) in `ValidateEgress` (for AWS as an example)